Mathlib/GroupTheory/Frattini.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2024 Colva Roney-Dougal. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Colva Roney-Dougal, Inna Capdeboscq, Susanna Fishel, Kim Morrison
+-/
+import Mathlib.Algebra.Group.Subgroup.Basic
+import Mathlib.Data.Fintype.Card
+import Mathlib.GroupTheory.Nilpotent
+import Mathlib.Order.Radical
+
+/-!
+# The Frattini subgroup
+
+We give the definition of the Frattini subgroup of a group, and three elementary results:
+* The Frattini subgroup is characteristic.
+* If every subgroup of a group is contained in a maximal subgroup, then
+the Frattini subgroup consists of the non-generating elements of the group.
+* The Frattini subgroup of a finite group is nilpotent.
+-/
+
+/-- The Frattini subgroup of a group is the intersection of the maximal subgroups. -/
+def frattini (G : Type*) [Group G] : Subgroup G :=
+  Order.radical (Subgroup G)
+
+variable {G H : Type*} [Group G] [Group H] {φ : G →* H} (hφ : Function.Surjective φ)
+
+lemma frattini_le_coatom {K : Subgroup G} (h : IsCoatom K) : frattini G ≤ K :=
+  Order.radical_le_coatom h
+
+open Subgroup
+
+lemma frattini_le_comap_frattini_of_surjective : frattini G ≤ (frattini H).comap φ := by
+  simp_rw [frattini, Order.radical, comap_iInf, le_iInf_iff]
+  intro M hM
+  apply biInf_le
+  exact isCoatom_comap_of_surjective hφ hM
+
+/-- The Frattini subgroup is characteristic. -/
+instance frattini_characteristic : (frattini G).Characteristic := by
+  rw [characteristic_iff_comap_eq]
+  intro φ
+  apply φ.comapSubgroup.map_radical
+
+/--
+The Frattini subgroup consists of "non-generating" elements in the following sense:
+
+If a subgroup together with the Frattini subgroup generates the whole group,
+then the subgroup is already the whole group.
+-/
+theorem frattini_nongenerating [IsCoatomic (Subgroup G)] {K : Subgroup G}
+    (h : K ⊔ frattini G = ⊤) : K = ⊤ :=
+  Order.radical_nongenerating h
+
+-- The Sylow files unnecessarily use `Fintype` (computable) where often `Finite` would suffice,
+-- so wwe need this:
```

Comment:
(Also, I'm going to try to clean up this fintype/finite mess in the group theory library, starting with #13431 and #13432, but eventually getting to p-groups and Sylow).
```suggestion
-- so we need this:
```